spec/services/idv/profile_step_spec.rb

```diff
@@ -68,62 +69,62 @@ def build_step(params)
         vendor: { reasons: nil },
       }
 
-      result = instance_double(FormResponse)
+      result = step.submit
 
-      expect(FormResponse).to receive(:new).
-        with(success: false, errors: errors, extra: extra).and_return(result)
-      expect(step.submit).to eq result
+      expect(result.success?).to eq(false)
+      expect(result.errors).to eq(errors)
+      expect(result.extra).to eq(extra)
       expect(idv_session.profile_confirmation).to be_nil
     end
 
     it 'fails with invalid first name' do
       step = build_step(user_attrs.merge(first_name: 'Bad'))
 
       errors = { first_name: ['Unverified first name.'] }
-
-      result = instance_double(FormResponse)
       extra = {
         idv_attempts_exceeded: false,
         vendor: { reasons: ['The name was suspicious'] },
       }
 
-      expect(FormResponse).to receive(:new).
-        with(success: false, errors: errors, extra: extra).and_return(result)
-      expect(step.submit).to eq result
+      result = step.submit
+
+      expect(result.success?).to eq(false)
+      expect(result.errors).to eq(errors)
+      expect(result.extra).to eq(extra)
       expect(idv_session.profile_confirmation).to be_nil
     end
 
     it 'fails with invalid ZIP code on current address' do
       step = build_step(user_attrs.merge(zipcode: '00000'))
 
       errors = { zipcode: ['Unverified ZIP code.'] }
-
-      result = instance_double(FormResponse)
       extra = {
         idv_attempts_exceeded: false,
         vendor: { reasons: ['The ZIP code was suspicious'] },
       }
 
-      expect(FormResponse).to receive(:new).
-        with(success: false, errors: errors, extra: extra).and_return(result)
-      expect(step.submit).to eq result
+      result = step.submit
+
+      expect(result.success?).to eq(false)
+      expect(result.errors).to eq(errors)
+      expect(result.extra).to eq(extra)
       expect(idv_session.profile_confirmation).to be_nil
     end
 
     it 'fails with invalid ZIP code on previous address' do
       step = build_step(user_attrs.merge(prev_zipcode: '00000'))
 
       errors = { zipcode: ['Unverified ZIP code.'] }
-
-      result = instance_double(FormResponse)
       extra = {
         idv_attempts_exceeded: false,
         vendor: { reasons: ['The ZIP code was suspicious'] },
       }
 
-      expect(FormResponse).to receive(:new).
-        with(success: false, errors: errors, extra: extra).and_return(result)
-      expect(step.submit).to eq result
+      result = step.submit
+
+      expect(result.success?).to eq(false)
+      expect(result.errors).to eq(errors)
+      expect(result.extra).to eq(extra)
       expect(idv_session.profile_confirmation).to be_nil
     end
 
```

Comment:
They did not pass because the `step.submit` would submit the form which would call `FormResponse.new` with different params than what the stub was looking for.